src/main/java/com/iexec/worker/dataset/DatasetService.java

```diff
@@ -34,12 +41,55 @@ public boolean downloadDataset(String chainTaskId, String datasetUri) {
                     chainTaskId, datasetUri);
             return true;
         }
-        return FileHelper.downloadFileInDirectory(datasetUri, getDatasetFolderPath(chainTaskId));
+        return FileHelper.downloadFileInDirectory(datasetUri, workerConfigurationService.getTaskInputDir(chainTaskId));
     }
 
-    public String getDatasetFolderPath(String chainTaskId) {
-        return configurationService.getResultBaseDir() + File.separator + chainTaskId + FileHelper.SLASH_INPUT;
+    public boolean decryptDataset(String chainTaskId, String datasetUri) {
+        String datasetFileName = Paths.get(datasetUri).getFileName().toString();
+        String datasetFilePath = workerConfigurationService.getTaskInputDir(chainTaskId) + File.separator + datasetFileName;
+        String datasetSecretFilePath = smsService.getDatasetSecretFilePath(chainTaskId);
+
+        if (!new File(datasetSecretFilePath).exists()) {
+            log.info("No dataset secret file found, will continue without decrypting dataset [chainTaskId:{}]", chainTaskId);
+            return true;
+        }
+
+        log.info("Decrypting dataset file [datasetFile:{}, secretFile:{}]", datasetFilePath, datasetSecretFilePath);
+
+        decryptFile(datasetFilePath, datasetSecretFilePath);
+
+        String decryptedDatasetFilePath = datasetFilePath + ".recovered";
+
+        if (!new File(decryptedDatasetFilePath).exists()) {
+            log.error("Decrypted dataset file not found [chainTaskId:{}, decryptedDatasetFilePath:{}]",
+                    chainTaskId, decryptedDatasetFilePath);
+            return false;
+        }
+
+        // replace original dataset file with decrypted one
+        return FileHelper.replaceFile(datasetFilePath, decryptedDatasetFilePath);
     }
 
+    public void decryptFile(String dataFilePath, String secretFilePath) {
+        String cmd = String.format("./decrypt-dataset.sh %s %s", dataFilePath, secretFilePath);
 
-}
+        ProcessBuilder pb = new ProcessBuilder(cmd.split(" "));
+        pb.directory(new File("./src/main/resources/"));
```

Comment:
We should parametrize it

Reply:
This is just temporary, we should remove those scripts and implement their logic in the worker codebase itself.